webportal/src/app/job/job-view/job-view.component.js

```diff
@@ -155,72 +154,78 @@ const convertGpu = (gpuAttribute) => {
-              convertTime(true, data[i].createdTime, data[i].completedTime),
-            retries: data[i].retries,
-            status: convertState(hjss),
-            stop: stopBtnStyle,
-          });
         }
-        $('#view-table').html(jobTableComponent({}));
-        table = $('#job-table').dataTable({
-          'data': displayDataSet,
-          'columns': [
-            {title: 'Job', data: 'jobName'},
-            {title: 'User', data: 'userName'},
-            {title: 'Virtual Cluster', data: 'vcName'},
-            {title: 'Start Time', data: 'startTime'},
-            {title: 'Duration', data: 'duration'},
-            {title: 'Retries', data: 'retries'},
-            {title: 'Status', data: 'status'},
-            {title: 'Stop', data: 'stop'},
-          ],
-          'scrollY': (($(window).height() - 265)) + 'px',
-          'lengthMenu': [[20, 50, 100, -1], [20, 50, 100, 'All']],
-          'order': [[3, 'desc']],
-          'columnDefs': [
-            {type: 'natural', targets: [0, 1, 2, 5, 6]},
-            {type: 'title-numeric', targets: [3, 4]},
-          ],
-          'deferRender': true,
-        }).api();
       }
-      loading.hideLoading();
-    },
-    error: (xhr, textStatus, error) => {
-      const res = JSON.parse(xhr.responseText);
-      alert(res.message);
-      loading.hideLoading();
     },
-  });
+    'columns': [
+      {title: 'Job', data: 'name', render(name, type) {
+        if (type !== 'display') return name;
+        return '<a href="view.html?jobName=' + name + '">' + name + '</a>';
+      } },
+      {title: 'User', data: 'username'},
+      {title: 'Virtual Cluster', data: 'virtualCluster', defaultContent: 'default'},
+      {title: 'Start Time', data: 'createdTime', render(createdTime, type) {
+        if (type !== 'display') return Math.round(createdTime / 1000);
+        return convertTime(false, createdTime)
+      }},
+      {title: 'Duration', data: null, render({createdTime, completedTime}, type) {
+        if (type !== 'display') return getDurationInSeconds(createdTime, completedTime);
+        return convertTime(true, createdTime, completedTime)
+      } },
+      {title: 'Retries', data: 'retries'},
+      {title: 'Status', data: null, render: getHumanizedJobStateString },
+      {title: 'Stop', data: null, render(job, type) {
+        let hjss = getHumanizedJobStateString(job);
+        return (hjss === 'Waiting' || hjss === 'Running') ?
+          '<button class="btn btn-default btn-sm" onclick="stopJob(\'' +
+            job.name + '\')">Stop</button>':
+          '<button class="btn btn-default btn-sm" disabled>Stop</button>';
+      } },
+    ],
+    'scrollY': (($(window).height() - 265)) + 'px',
+    'lengthMenu': [[20, 50, 100, -1], [20, 50, 100, 'All']],
+    'order': [[3, 'desc']],
+    'columnDefs': [
+      {type: 'natural', targets: [0, 1, 2, 5, 6]},
+      {type: 'num', targets: [3, 4]},
+    ],
+    'deferRender': true,
+    'autoWidth': false,
```

Comment:
What does autoWidth mean and what will happen if it is dialsabled?

Reply:
When autoWidth is enabled, DataTables will layout all possible rows (which means 350,000 rows in the benchmarks). collect widths of every cell and calculate the most suitable width of each column. When it is disabled, widths of each column will be decided by the browser, based on data in the current page. So this will cause different page have different column widths (but total will remain 100% of table width, which is set explicitly in CSS)

Reply:
Thanks! Sounds good.